### ISSUE TITLE
home ui restyle and more...

### ISSUE THREAD

**Issue by carmelopellegrino** [MechanicalVentilatorMilano/gui#119]
first of all the more...
- [x] the reply to the `get all` command contains the following fields, in order of appearance:
1. instantaneous pressure  (plot)
1. instantaneous volume (plot)
1. o2 (monitor and alarm)
1. bpm (monitor and alarm)
1. tidal volume (plot)
1. PEEP (monitor and alarm)
1. temperature (ignore)
1. battery_powered (0 = power from 220V, 1 = power from the battery)
1. battery charge (monitor, in %)
1. PEAK (monitor and alarm)
1. total inspired volume (monitor and alarm)
1. totale expired volume (monitor and alarm)
1. volume minute (monitor and alarm)

the following restyle of the home ui has been requested:
- [x] the monitor shall report only a label, the value and the unit (i.e. no limits, no mean, no max, etc...) The monitored values change very slowly as compared to the plotted ones

- [x] the following observables (8 in total) go on the right monitor column:
- o2 (monitor and alarm)
- bpm (monitor and alarm)
- PEEP (monitor and alarm)
- battery charge (monitor, in %) and battery_powered  
- PEAK (monitor and alarm)
- total inspired volume (monitor and alarm)
- totale expired volume (monitor and alarm)
- volume minute (monitor and alarm)

- [x] alarms are only on those observables where it is reported, i.e. the plotted values do not trigger any alarm. Not all the alarms have a minimal threshold, not all the alarms have a maximal. Up to now, I know that PEEP has a minimal and PEAK has a maximal

**Comment by carmelopellegrino** [comment]
note for the future: here is the updated `get_all_fields` list to replace the current one with.

```yaml
get_all_fields:
  - instant_pressure
  - instant_volume
  - oxygen
  - bpm
  - tidal_volume
  - peep
  - temperature
  - battery_powered
  - battery_charge
  - peak
  - total_inspired_volume
  - total_expired_volume
  - volume_minute
```

**Comment by carmelopellegrino** [comment]
New list of constants for the HW alarms:

```c++
ERROR_INPUT_PRESSURE_LOW      1<<0
ERROR_INPUT_PRESSURE_HIGH     1<<1
ERROR_INSIDE_PRESSURE_LOW     1<<2
ERROR_INSIDE_PRESSURE_HIGH    1<<3
ERROR_BATTERY_LOW             1<<4
ERROR_LEAKAGE                 1<<5
ERROR_FULL_OCCLUSION          1<<6
ERROR_PARTIAL_OCCLUSION       1<<7
ERROR_ALARM_PI                1<<29
ERROR_WDOG_PI                 1<<30
ERROR_SYSTEM_FALIURE          1<<31
```

There are 2 new items: `ERROR_ALARM_PI` and `ERROR_WDOG_PI`.

**Comment by carmelopellegrino** [comment]
the parameter to be set (and read) for the support pressure parameter is `pressure_support`.

**Comment by carmelopellegrino** [comment]
- [ ] differentiate between GUI and HW alarms in `ESP32Serial`.

The pattern is:
- Case1: an HW alarm raises. The GUI gets to know by repeatedly calling `ESP32Serial.get_alarms()`. The GUI shows the error and provides a "snooze" button. When this button is clicked, the GUI sends to the ESP the command `set alarm_snooze <exponent>` where `<exponent>` is a placeholder for the exponent to be given to the power of 2 to get the alarm code number.
- Case 2: the GUI triggers an alarm on input values from the ESP monitored data. The GUI makes the corresponding monitor red and tells to the ESP that there is an alarm condition by issuing `set alarm 1`. Internally, the ESP code reacts to this message by raising the bit 29 up in the number returned by the `ESP32.get_alarms()` GUI function. By clicking on the corresponding red monitor, the command `set alarm_snooze 29` shall be set to the ESP.

**Comment by carmelopellegrino** [comment]
solved in PR #130
